tests/integration/__init__.py

```diff
@@ -1,23 +1,167 @@
 import os
+from typing import Union
 
 from tests import TestBase
 
-
 class IntegrationTestBase(TestBase):
+    pass
+
+class ApplicationTestBase(TestBase):
     """
-    Base class for express integration tests.
+    Base class for express application parser tests.
     """
 
-    def setUp(self):
-        super(IntegrationTestBase, self).setUp()
+    def __init__(
+        self,
+        *args,
+        application=None,
+        version=None,
+        subdir=None,
+        filename=None,
+        property=None,
+        test_config=None,
+        **kwargs
+    ):
+        super(ApplicationTestBase, self).__init__(*args, **kwargs)
+        self.application = application
+        self.version = version
+        self.subdir = subdir
+        self.filename = filename
+        self.property = property
+        self.test_config = test_config
+
+    def work_dir(self, version: str, subdir: str):
+        return os.path.join(
+            self.rootDir, "fixtures", self.application, version, subdir
+        )
+
+    def get_reference_value(self):
+        """Obtain the appropriate reference value to use as an expected value for the
```

Comment:
Here and elsewhere in the PR - let's keep the newlines for the docstrings, pls, to be consistent with previous styling.

Reply:
Done, but we should really just use an auto-formatter for this kind of stuff.